wordpress-mobile/WordPress-iOS
# Update Core Data model minimum tools version

This PR updates the minimum "Tools Version" used by our current Core Data model **WordPress 97.xcdatamodel** from Xcode 7.3 to 9.0.

### Rationale

This was the minimum change necessary to enable the (newish) Core Data attribute type, URI, which allows URLs to be modeled directly as `URL` instead of as `String`. Without this change, using URI produced an error:

<img width="1000" alt="Error caused by old model version" src="https://user-images.githubusercontent.com/1898325/90576576-9f269b80-e18c-11ea-9a33-024d5694c978.png">

### Changes

Updating the version changes the underlying XML data model automatically as seen in the diff. These are the changes I see:
- `minimumToolsVersion="Xcode 7.3"` becomes `minimumToolsVersion="Xcode 9.0"`
- Changes related to indexing:
  - Removal of `indexed="YES"` from model attributes
  - addition of `<fetchIndex>`, [which is related to a new indexing API](https://stackoverflow.com/a/47852563/1305067)


### Testing

This change doesn't involve a new Core Data model version, but I think it's important to test a simple app update scenario. I tested using the following steps:
1. Install the app on a device using the current `develop` branch
2. Log in to a site and perform some common actions such as updating a post
3. Switch over to this branch and re-run the app on the device
4. Open the app, open the post and ensure there's no crash

PR submission checklist:
